lib/chef/knife/vsphere_vm_snapshot.rb

```diff
@@ -147,17 +147,23 @@ def find_node(tree, name)
   end
 
   def display_node(node, current, shift = 1)
-    descr = node.name + ' ' + node.createTime.iso8601
-    out = ''
-    out << '+--' * shift
+    snapshot_tree = {}
+    children = []
+    unless node.childSnapshotList.empty?
+      node.childSnapshotList.each { |item| children << display_node(item, current, shift + 1) }
+    end
     if node.snapshot == current
-      out << ui.color(descr, :cyan) << '\n'
+      snapshot_tree = { 'SnapshotName' => node.name,
+                       'SnapshotDescription' => node.description,
+                       'SnapshotCreationDate' => node.createTime.iso8601,
+                       'IsCurrentSnapshot' => true,
+                       'Children' => children }
     else
-      out << descr << '\n'
-    end
-    unless node.childSnapshotList.empty?
-      node.childSnapshotList.each { |item| out << display_node(item, current, shift + 1) }
+      snapshot_tree = { 'SnapshotName' => node.name,
+                       'SnapshotDescription' => node.description,
+                       'SnapshotCreationDate' => node.createTime.iso8601,
+                       'Children' => children }
     end
-    out
+    return snapshot_tree
```

Comment:
This return should be redundant since the last assignment is to `snapshot_tree` and Ruby will use that.

Reply:
yes, I realise this, but coming from a Java background, it was an automatic thing to do :P  I can make the change :)

Reply:
I have successfully removed the snapshot_tree initialization.  However, when I removed the return statement, the method returns null - do you have any suggestions around this?

![image](https://user-images.githubusercontent.com/23378716/37552320-74db4a30-29fe-11e8-9355-9db59b7f5059.png)

Reply:
Ah, sorry, I think I gave you some conflicting advice. If you ended up going with

```
snapshot_tree = { blah }
snapshot_tree.merge! { blah } if something
```

Then follow that up with a bare `snapshot_tree` rather than `return snapshot_tree`.

Typically in Ruby the `return` keyword is only used if you're returning from the method early. If you get to the end of the method the last statement is used as an implicit return. (I swear it'll start making sense, stick with it ;) )

Reply:
Silly example to show what I meant:

```Ruby
irb(main):012:0> def foo(boolean)
irb(main):013:1> test = { name: 'Sean' }
irb(main):014:1> test.merge!({ occupation: 'Engineer' }) if boolean
irb(main):015:1> end
=> :foo
irb(main):016:0> foo(false)
=> nil
irb(main):017:0> foo(true)
=> {:name=>"Sean", :occupation=>"Engineer"}
```

Reply:
haha...ok I'll do what you said...it just looks and feels weird not to put a return statement let alone just writing the object with explicit 'return'...I'll need to get used to this one :D